packages/core/src/components/overlay2/overlay2.tsx

```diff
@@ -408,7 +408,6 @@ export const Overlay2 = React.forwardRef<OverlayInstance, Overlay2Props>((props,
                 // IMPORTANT: only inject our ref if the user didn't specify childRef or childRefs already. Otherwise,
                 // we risk clobbering the user's ref (which we cannot inspect here while cloning/decorating the child).
                 ref: userChildRef === undefined ? localChildRef : undefined,
-                tabIndex: enforceFocus || autoFocus ? 0 : undefined,
```

Comment:
This `tabIndex={0}` doesn't do anything because it's not on an interactive element. And `autoFocus` brings focus to `startFocusTrapElement`, not this element.